src/pages/admin-settings.tsx

```diff
@@ -0,0 +1,319 @@
+  const [taskName, setTaskName] = useState('');
+  const [assignedUsers, setAssignedUsers] = useState('');
+
+  const handleUpdateTask = () => {
+    // Update task information in the database
+    setShowModal(false);
+  };
+
+  const handleCancel = () => {
+    setShowModal(false);
+  };
+
+  return (
+    <div>
+      <h2 className='text-2xl font-bold mb-4'>Create a New Task</h2>
+      <label className='block mb-2'>
+        Task Name:
+        <input
+          className='border border-gray-400 rounded w-full p-2'
+          type='text'
+          value={taskName}
+          onChange={(e) => setTaskName(e.target.value)}
+        />
+      </label>
+      <label className='block mb-4'>
+        Assigned Users:
+        <input
+          className='border border-gray-400 rounded w-full p-2'
+          type='text'
+          value={assignedUsers}
+          onChange={(e) => setAssignedUsers(e.target.value)}
+        />
+      </label>
+      <button
+        className='bg-purple-600 text-white py-2 px-4 rounded mr-2'
+        onClick={handleUpdateTask}
+      >
+        Update Task
+      </button>
+      <button
+        className='bg-gray-200 text-red-600 py-2 px-4 rounded'
+        onClick={handleCancel}
+      >
+        Cancel
+      </button>
+    </div>
+  );
+};
+
+const Users = () => {
+  const [displaySkills, setDisplaySkills] = useState<number | undefined>(undefined);
+  const [showModal, setShowModal] = useState(false);
+  const [currentTable, setCurrentTable] = useState(1);
+  const [modalContent, setModalContent] = useState<React.ReactNode>(null);
+  const [showSortOptions, setShowSortOptions] = useState(false);
+
+
+const handleMoreClick = (id: number) => {
+    if (displaySkills === id) {
+      setDisplaySkills(undefined);
+    } else {
+      setDisplaySkills(id);
+    }
+  };
+  
+
+  return (
+    <div className='bg-gray-100 min-h-screen'>
+      {showModal && (
+      <div className='fixed top-0 left-0 w-full h-full bg-black bg-opacity-50 flex justify-center items-center'>
+        <div className='bg-white p-4 rounded-lg'>
+          {modalContent}
+          </div>
+          </div>
+          )}
+      <div className='flex justify-between p-4'>
+        <h2>Admin Dashboard</h2>
+        <h2>Welcome Back, Nhi</h2>
+      </div>
+      <div className='p-4'>
+  <div className='flex justify-between mb-4'>
+    <div>
+      <button
+        className={`py-2 px-4 rounded ${
+          currentTable === 1 ? 'bg-purple-600 text-white' : 'bg-gray-200 text-purple-800'
+        }`}
+        onClick={() => setCurrentTable(1)}
+      >
+        Users
+      </button>
+      <button
+        className={`py-2 px-4 rounded ml-2 ${
+          currentTable === 3 ? 'bg-purple-600 text-white' : 'bg-gray-200 text-purple-800'
+        }`}
+        onClick={() => setCurrentTable(3)}
+      >
+        Task Queues
+      </button>
+    </div>
+    {currentTable === 3 && (
```

Comment:
What do you think about making separate components for the Users and Task Queues table?